docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,230 @@
+
+**<i>TODO: Explanation of what is required from Substreams module perspective. We need to talk here about the Protobuf gen required to pull sf.substreams.sink.files.v1 model and modifications required, e.g. outputting a JSON model for each entity, one entity per line.</i>**
+
+Run the included `make codegen` command to create the required protobuf files.
+
+```bash
+make codegen
+```
+
+It's a good idea to run the example code using your installation of the `substreams` CLI to make sure everything is set up and working properly.
+
+Verify the setup for the example project by using the `make build` and `substreams run` commands.
+
+Build the Substreams module by using the included `make` command.
+
+```bash
+make
+```
+
+Run the project by using the `substreams run` command.
+
+```bash
+substreams run -e mainnet.eth.streamingfast.io:443 substreams.yaml jsonl_out --start-block 1000000 --stop-block +1
+```
+
+The `substreams run` command will result in output resembling the following:
+
+```bash
+----------- NEW BLOCK #1,000,000 (1000000) ---------------
+{
+  "@module": "jsonl_out",
+  "@block": 1000000,
+  "@type": "sf.substreams.sink.files.v1",
+  "@data": {
+    "lines": [
+      "{\"hash\":\"8e38b4dbf6b11fcc3b9dee84fb7986e29ca0a02cecd8977c161ff7333329681e\",\"number\":1000000,\"parent_hash\":\"b4fbadf8ea452b139718e2700dc1135cfc81145031c84b7ab27cd710394f7b38\",\"timestamp\":\"2016-02-13T22:54:13Z\"}"
+    ]
+  }
+}
+```
+
+## Substreams modifications
+
+### Module handler changes for sink
+
+The example code in the [`lib.rs`](#) Rust source code file contains the `jsonl_out` module handler responsible for extracting the blockchain data. The module handler is responsible for passing the data to the `sf.substreams.sink.files.v1` protobuf for the sink tool and its processes.
+
+```rust
+#[substreams::handlers::map]
+fn jsonl_out(block: eth::Block) -> Result<Lines, substreams::errors::Error> {
+
+    let header = block.header.as_ref().unwrap();
+
+    Ok(pb::sinkfiles::Lines {
+        lines: vec![json!({
+            "number": block.number,
+            "hash": Hex(&block.hash).to_string(),
+            "parent_hash": Hex(&header.parent_hash).to_string(),
+            "timestamp": header.timestamp.as_ref().unwrap().to_string()
+        })
+        .to_string()],
+    })
+}
+```
+
+The module handler uses `JSONL` for the output type, `CSV` is also supported.. The [`json!`](https://docs.rs/serde_json/latest/serde_json/macro.json.html) macro is used to write the block data to the Rust `Vec` type by using the Rust [`vec!`](https://doc.rust-lang.org/std/macro.vec.html) macro.
+
+**TODO**: <i>We just need to emphasis in the tutorial that it's an example and that a lot of "rows" can be extracted per block. Giving the ERC20/ERC721/ERC1155 examples is a good idea as we coded it: https://github.com/streamingfast/substreams-eth-token-transfers/blob/develop/src/lib.rs#L24</i>
```

Comment:
Do you need more info for this?

Reply:
Yes, please. Anything you can provide here that might be helpful to the reader is greatly appreciated.

Reply:
I'm not sure what more to put, I just want to ensure that user don't think that because in the sample we have one block == one line, it's a requirement. Just want to add a quick sentence that explain that many entities can be extracted from a single block. Redirecting user to eth-token-transfers would be given at the end something like "To see a real-world example of extracting multiple entities from a single block, use can refer to [substreams-eth-token-transfers](...) to see how N transfers can be extracted from a single block and sent back into multiple lines, one line per transfer"